src/DataViews/SamplingReportDataView.cpp

```diff
@@ -315,10 +331,14 @@ void SamplingReportDataView::SetSampledFunctions(const std::vector<SampledFuncti
 
 void SamplingReportDataView::SetThreadID(ThreadID tid) {
   tid_ = tid;
+  const orbit_client_data::CaptureData& capture_data = app_->GetCaptureData();
+
   if (tid == orbit_base::kAllProcessThreadsTid) {
-    name_ = absl::StrFormat("%s\n(all threads)", app_->GetCaptureData().process_name());
+    name_ = absl::StrFormat("%s\n(all threads)", capture_data.process_name());
+    thread_events_count_ = capture_data.GetCallstackData().GetCallstackEventsCount();
   } else {
-    name_ = absl::StrFormat("%s\n[%d]", app_->GetCaptureData().GetThreadName(tid_), tid_);
+    name_ = absl::StrFormat("%s\n[%d]", capture_data.GetThreadName(tid_), tid_);
+    thread_events_count_ = capture_data.GetCallstackData().GetCallstackEventsOfTidCount(tid_);
```

Comment:
The way an instance of this class is created filled is a bit weird already, but still, it feels a bit strange to initialize `thread_events_count_` here like this. I would add another setter, and pass in `ThreadSampleData::samples_count` (see `SamplingReport::FillReport`) instead of retrieving it. Improving the constructor is also a possibility but that's a bit of a broader scope already. Maybe someone else has a different opinion, this class is _very_ old.

Reply:
Ok well, maybe this is actually consistent with how we get the thread name. Not sure.

Reply:
> consistent with how we get the thread name
that's exactly what I've replicated.

Reply:
Yeah I guess that's fine and it's a consequence of the existing code.